main/distributed-programming.md

```diff
@@ -94,88 +125,126 @@ harden(o);
-offers (reallocations and completions), and by some contracts, which can publish
-current prices or other details specific to the contract.
+Dapps and other tools may want to be notified about state changes of a Zoe
+contract or offer. Agoric uses a Notifier based on Promises, allowing
+many subscribers to receive notifications without the publisher 
+having to track a subscription list.
 
-#### Getting notifications
+Zoe supports the Notifier, which publishes updates to offer state
+(reallocations and completions). Some contracts also use it, and
+can publish current prices or other contract-specific details
 
-Zoe has a public method `getOfferNotifier()`, and contracts will have similar
-methods. This provides a long-lived notifier object associated with a particular
-stream of updates. At any time, you can call `notifier.getUpdateSince()`, and it will
-return a record `{ value, updateHandle, done }`. `value` is the current state,
-according to the source. `done` will be false until the stream reaches a final state,
-after which `value` will never change, and the same record will always be
-returned. `updateHandle` allows you to ask to be notified the next time there's a
-change to the state.
+### Getting notifications
 
-If you call `getUpdateSince(oldUpdateHandle)` with no handle, or any `updateHandle`
-other than the most recent one, the notifier will immediately return a record with
-the current state. If you call `getUpdateSince(updateHandle)` with the most-recently
-generated `updateHandle`, you'll get a promise for the next record, which will be
-resolved the next time there's a change in the state.
+Zoe has a public method `getOfferNotifier()`, and contracts will have
+similar methods. This method provides a long-lived notifier object associated 
+with a particular stream of updates.
 
-Well-designed notification systems can provide access either to the most recent
-state, or to a complete list of changes. (Sometimes providing changes since a recent
-epoch improves the UX.) This API is designed to support the former style of update,
-though it's a simple work-around to include deltas as part of the state.
+```js
+const offerNotifer = zoe.getOfferNotifier(offerHandle);
+  const { value, updateHandle, done } = offerNotifier.getUpdateSince();
+  if (done) {
+   <drop offer from list>
+  }
+  newValue = value;
+  waitForNextUpdate(offerNotifier, updateHandle);
+```
 
-A common pattern for following updates to the notifier until it's done looks like this:
+Note: There is both a `zoe.getOfferNotifier()` and a `zcf.getOfferNotifier()`. Use the `zcf.` version 
+within contracts and the `zoe.` version in the REPL, deploy scripts, and similar outside of a contract cases. 
+
+When called on a notifier object`notifier.getUpdateSince()` returns
+the record `{ value, updateHandle, done }`. `value` represents 
+the state of an offer or contract. If you get a notifier from Zoe, you have to 
+identify the offer. 
```

Comment:
"If you get a notifier from Zoe, you have to identify the offer. "
should be either

"If you want a notifier from Zoe, you have to identify the offer. "
or 
"To get a notifier from Zoe, you have to identify the offer. "